### PR TITLE
Roadmap

### PR DESCRIPTION
**Pull request by StephanEwen**
This adds a roadmap to the website.

The roadmap was proposed and discussed in https://lists.apache.org/thread.html/c1e755cf11214529546d01b6f4eacb756879cd8137f480760ce25580@%3Cdev.flink.apache.org%3E

The main file is https://github.com/StephanEwen/flink-web/blob/e1fea3c92b0892d68fb2ae8cdd0dbd56493b3cd3/roadmap.md

### CODE REVIEW

File: roadmap.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+<hr />
+
+{% toc %}
+
+**Preamble:** This is not an authoritative roadmap in the sense of a strict plan with a specific
+timeline. Rather, we, the community, share our vision for the future and give an overview of the bigger
+initiatives that are going on and are receiving attention. This roadmap shall give users and
+contributors an understanding where the project is going and what they can expect to come.
+
+The roadmap is continuously updated. New features and efforts should added to the roadmap once
```
Comment (sunjincheng121) [inline roadmap.md]: `should added` -> `should be added` ?

File: roadmap.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+there is consensus that they will happen and what they will roughly look like for the user.
```
Comment (sunjincheng121) [inline roadmap.md]: `consensus` -> `the consensus` ?

File: roadmap.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+# Analytics, Applications, an the roles of DataStream, DataSet, and Table API
+
+Flink views stream processing as a [unifying paradigm for data processing]({{ site.baseurl }}/flink-architecture.html)
+(batch and real-time) and event-driven applications. The APIs are evolving to reflect that view:
+
+  - The **Table API / SQL** is becoming the primary API for analytical use cases, in a unified way
+    across batch and streaming. To support analytical use cases in a more streamlined fashion,
+    the API is [extended with additional functions](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=97552739)
+
+    Like SQL, the Table API is *declarative*, operates on *logical schema*, and applies *automatic optimization*.
```
Comment (sunjincheng121) [inline roadmap.md]: `logical schema` -> `a logical schema`?

File: roadmap.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+there is consensus that they will happen and what they will roughly look like for the user.
+
+# Analytics, Applications, an the roles of DataStream, DataSet, and Table API
+
+Flink views stream processing as a [unifying paradigm for data processing]({{ site.baseurl }}/flink-architecture.html)
+(batch and real-time) and event-driven applications. The APIs are evolving to reflect that view:
+
+  - The **Table API / SQL** is becoming the primary API for analytical use cases, in a unified way
+    across batch and streaming. To support analytical use cases in a more streamlined fashion,
+    the API is [extended with additional functions](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=97552739)
```
Comment (sunjincheng121) [inline roadmap.md]: `the API is [extended with additional functions]` -> `the API  enhancement [FLIP-29](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=97552739) that extended with additional functions.`

File: roadmap.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+  - External Shuffle Services (mainly bounded streams) to support decoupling from computation and
+    intermediate results for better resource efficiency on systems like Yarn.
+    [FLIP-31](https://cwiki.apache.org/confluence/display/FLINK/FLIP-31%3A+Pluggable+Shuffle+Manager).
+
+Various of these enhancements can be taken form the contributed code from the
+[Blink fork](https://github.com/apache/flink/tree/blink).
+
+To exploit the above optimizations for bounded streams in the DataStream API, we need
+break parts of the API and explicitly model bounded streams.
```
Comment (sunjincheng121) [inline roadmap.md]: `we need break` -> `we need to break`?

File: roadmap.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+better introspection of running jobs [FLINK-10705](https://issues.apache.org/jira/browse/FLINK-10705).
+
+The community is working on extending the interoperability with authentication and authorization services.
+Under discussion are general extensions to the [security module abstraction](http://apache-flink-mailing-list-archive.1008284.n3.nabble.com/DISCUSS-Flink-security-improvements-td21068.html)
+as well as specific [enhancements to the Kerberos support](http://apache-flink-mailing-list-archive.1008284.n3.nabble.com/DISCUSS-Flink-Kerberos-Improvement-td25983.html).
+
+# SQL and Table API Ecosystem
+
+The community is working on extending the support for catalogs, schema registries, and
+metadata stores, including support in the APIs and the SQL client.Registries support:
```
Comment (sunjincheng121) [inline roadmap.md]: `client.Registries ` ->` client. Registries ` add a space.

File: roadmap.zh.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
+---
+title: "Apache Flink Roadmap"
```
Comment (sunjincheng121) [inline roadmap.zh.md]: We may write this document in Chinese? (We can also translate this document into Chinese in a separate PR.)

File: _data/i18n.yml [hunk 8]
```diff
@@ -23,4 +24,4 @@ zh:
     flink_blog: Flink 博客
     community_project: 社区 &amp; 项目信息
     how_to_contribute: 如何参与贡献
-
+    roadmap: Roadmap
```
Comment (sunjincheng121) [inline _data/i18n.yml]: `Roadmap` -> `路线图 or 蓝图` ？（We can also translate this into Chinese in a separate PR. together with `roadmap.zh.md`) :-)

File: roadmap.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+[FLINK-11275](https://issues.apache.org/jira/browse/FLINK-11275).
+Adding tables and streams to the catalogs via DDL is tracked via
+[FLINK-10232](https://issues.apache.org/jira/browse/FLINK-10232).
+
+There is a broad effort to integration Flink with the Hive Ecosystem, including
+metastore andHIve UDF support [FLINK-10556](https://issues.apache.org/jira/browse/FLINK-10556).
```
Comment (KurtYoung) [inline roadmap.md]: and hive

File: _data/i18n.yml (hunk 8 above)
Comment (wuchong) [reply _data/i18n.yml]: I have created an issue for this [FLINK-11754](https://issues.apache.org/jira/browse/FLINK-11754)  ;-)

Comment (sunjincheng121) [reply _data/i18n.yml]: Sounds good! Thanks @wuchong

**Comment by shaoxuan-wang** [comment]
@StephanEwen thanks for spending time to write this great roadmap. 
In the "fast batch" section, it seems you only have mentioned the on-going runtime improvements. It would be great if you can add a few outlines to include the improvements on the "Query execution: execution optimizations including binary row, hot method and operator codegen", "Query optimization: a state-of-art query planner with many new advanced rules"

**Comment by StephanEwen** [comment]
Thanks for the comments and reviews! Will fix the typos and errors as suggested.

@shaoxuan-wang Let's add "Improved SQL Query Processor", with the aspects you mentioned. I don't think we have to go into the details, it should be enough to describe that this leads to significant better SQL coverage and much improved performance. Do you have an umbrella JIRA issues / FLIP / design doc that I could link?

File: roadmap.md (hunk 1 above)
Comment (StephanEwen) [reply roadmap.md]: will fix

File: roadmap.md (hunk 2 above)
Comment (StephanEwen) [reply roadmap.md]: I think this is correct as it is

File: roadmap.md (hunk 3 above)
Comment (StephanEwen) [reply roadmap.md]: will fix

File: roadmap.zh.md (hunk 7 above)
Comment (StephanEwen) [reply roadmap.zh.md]: definitely :-)

**Comment by shaoxuan-wang** [comment]
> Thanks for the comments and reviews! Will fix the typos and errors as suggested.
> 
> @shaoxuan-wang Let's add "Improved SQL Query Processor", with the aspects you mentioned. I don't think we have to go into the details, it should be enough to describe that this leads to significant better SQL coverage and much improved performance. Do you have an umbrella JIRA issues / FLIP / design doc that I could link?

Yes, you are right. The design details of QP have not been clearly stated and discussed. I think all of these improvements will be merged in the umbrella JIRA of FLIP32: [FLINK-11439](https://issues.apache.org/jira/browse/FLINK-11439)

**Comment by sunjincheng121** [comment]
Thanks for the update @StephanEwen!
LGTM. +1 to merged.
Best,
Jincheng

**Comment by zentol** [comment]
@StephanEwen I presume this PR can be closed?

**Comment by fhueske** [comment]
PR was merged. Closing.

**Pull request closed without merging**